### PR TITLE
refactor: implementation of locale calculated server-side

### PR DESCRIPTION
**Pull request by grzpab**
## What does this PR do?

This issue moves the calculation of the current user's locale into the backend and ensures there is just one way of calculating it.
The algorithm is as follows:
1. if the user token is present in the cookies, extract the locale by decrypting the JWT/JWE token,
2. if the user token is present in the authorization header, extract the locale by decrypting the JWT/JWE token,
3. if the user token is not present, extract the locale from the `accept-language` header,
4. otherwise the locale is `en`,

The application does not longer rely on `router.locale` or similar, instead we pass `newLocale` calculated in `_document` and made available to the frontend code by embedding it as `window.calNewLocale`.
We obtain the `newLocale` again in `getInitialProps` within `_app` using:
1. `window.calNewLocale` for the frontend
2. the request object available in the backend
and we pass it to the components using the PageWrapper / AppProviders utilities.

We removed the `I18nLanguageHandler`.

When you change the language in the settings, you will be logged out. The reason for this behavior is that without getting a new token, we would still read the old language from the current token. We are working on redeeming the token when changing the language so the UX is better.

Fixes # (issue)
No fixes.

## Requirement/Documentation
https://github.com/calcom/cal.com/issues/11437

## Type of change
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?
This change requires a whole test of the application, especially ensuring that:
* the locale is always available and correct translations are visible at all times
* the locale stays the same when moving between different pages
* there are no changes between different locale during the page loading (which happens currently if there are discrepancies between the server-side locale and client-side locale)

## Mandatory Tasks
I did review the current PR.

## Checklist

- I haven't checked if my PR needs changes to the documentation
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by alex-from-intuita** [comment]
@grzpab

File: packages/features/auth/lib/getServerSession.ts [hunk 1]
```diff
@@ -13,17 +13,6 @@ import prisma from "@calcom/prisma";
...
- * This is a slimmed down version of the `getServerSession` function from
- * `next-auth`.
- *
- * Instead of requiring the entire options object for NextAuth, we create
- * a compatible session using information from the incoming token.
- *
- * The downside to this is that we won't refresh sessions if the users
- * token has expired (30 days). This should be fine as we call `/auth/session`
- * frequently enough on the client-side to keep the session alive.
- */
```
Comment (zomars) [inline packages/features/auth/lib/getServerSession.ts:26]: Is this intended?

File: packages/trpc/server/routers/publicViewer/i18n.handler.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ type I18nOptions = {
   input: I18nInputSchema;
 };
 
-export const i18nHandler = async ({ input }: I18nOptions) => {
+export const i18nHandler = async ({ ctx, input }: I18nOptions) => {
```
Comment (zomars) [inline packages/trpc/server/routers/publicViewer/i18n.handler.ts:14]: Is ctx being used here?

**Review by zomars — COMMENTED** [review]
Looking good so far. Just noticing some typing errors. Will re-review once those are cleared out. 🙏🏽

File: packages/features/auth/lib/getServerSession.ts (hunk 1 above)
Comment (grzpab) [reply packages/features/auth/lib/getServerSession.ts]: You are right, I restored the comment.

File: packages/trpc/server/routers/publicViewer/i18n.handler.ts (hunk 2 above)
Comment (grzpab) [reply packages/trpc/server/routers/publicViewer/i18n.handler.ts]: It is not, I removed the usage of it and removed it from the `I18nOptions` type.

File: yarn.lock
Comment (zomars) [inline yarn.lock:1]: Can we remove this change from the PR? Everything else is fine and I can see checks are passing now. 🙏🏽

Comment (grzpab) [reply yarn.lock:1]: @zomars done, thanks for reviewing.

**Review by zomars — APPROVED** [review]

**Comment by zomars** [comment]
Heads up @grzpab this PR is causing spacing issues on main since we're not setting the html `dir="ltr"` attribute anymore. We're working on a fix.

**Pull request merged**